packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx

```diff
@@ -0,0 +1,228 @@
+    ? YPP_AUTHORIZATION_STEPS
+    : YPP_AUTHORIZATION_STEPS_WITHOUT_CHANNEL_SELECT // if the user has only a single channel, skip "Select channel" step
+  const currentStep = currentStepIdx != null ? authorizationSteps[currentStepIdx] : null
+
+  const goToLoadingStep = useCallback(() => {
+    setCurrentStepIdx(authorizationSteps.findIndex((step) => step === 'fetching-data'))
+  }, [authorizationSteps, setCurrentStepIdx])
+
+  const { handleAuthorizeClick } = useYppGoogleAuth({
+    closeModal: useCallback(() => setCurrentStepIdx(null), [setCurrentStepIdx]),
+    channelsLoaded,
+    goToLoadingStep,
+    selectedChannelId,
+    setSelectedChannelId,
+    setCurrentStepIdx,
+  })
+
+  const selectedChannel = useMemo(() => {
+    if (!channels || !selectedChannelId) {
+      return null
+    }
+    return channels.find((channel) => channel.id === selectedChannelId)
+  }, [channels, selectedChannelId])
+  const isSelectedChannelValid = useMemo(
+    () =>
+      (selectedChannel &&
+        !!selectedChannel.avatarPhoto &&
+        !!selectedChannel.coverPhoto &&
+        !!selectedChannel.description) ??
+      false,
+    [selectedChannel]
+  )
+
+  const goToNextStep = useCallback(
+    () => setCurrentStepIdx((prev) => (prev != null ? prev + 1 : null)),
+    [setCurrentStepIdx]
+  )
+  const goToPreviousStep = useCallback(
+    () => setCurrentStepIdx((prev) => (prev != null ? prev - 1 : null)),
+    [setCurrentStepIdx]
+  )
+
+  // if selected channel is not set, default to the first one
+  useEffect(() => {
+    if (!channels || !channels.length || selectedChannelId) {
+      return
+    }
+    setSelectedChannelId(channels[0].id)
+  }, [channels, selectedChannelId, setSelectedChannelId])
+
+  const authorizationStep = useMemo(() => {
+    switch (currentStep) {
+      case 'select-channel':
+        return {
+          title: 'Select channel',
+          description: 'Select the Atlas channel you want your YouTube channel to be connected with.',
+          primaryButton: {
+            text: 'Select channel',
+            onClick: goToNextStep,
+            disabled: !selectedChannel,
+          },
+          component: (
+            <YppAuthorizationSelectChannelStep
+              channels={channels}
+              selectedChannelId={selectedChannelId}
+              onSelectChannel={setSelectedChannelId}
+            />
+          ),
+        }
+      case 'requirements':
+        return {
+          title: 'Requirements',
+          description:
+            'Before you can apply to the program, make sure both your Atlas and YouTube channels meet the below conditions.',
+          primaryButton: {
+            text: 'Authorize with YouTube',
+            onClick: handleAuthorizeClick,
+            disabled: !isSelectedChannelValid,
+          },
+          component: (
+            <YppAuthorizationRequirementsStep
+              onChangeChannel={() => selectedChannel && setActiveUser({ channelId: selectedChannel?.id })}
```

Comment:
We could either omit `?.` here or remove `selectedChannel &&`

Reply:
Fixed